### ISSUE TITLE
[FR] Add step argument to log_figure

### ISSUE THREAD

**Issue by nlgranger** [mlflow/mlflow#6261]
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

Add step argument to mlflow.log_figure()

Add slider in ui to navigate over time/steps.

### Motivation

> #### What is the use case for this feature?

It is currently impossible to analyze the dynamic of metrics visualized via figures (histograms, confusion matrices, roc curves, model outputs, etc.)

> #### Why is this use case valuable to support for MLflow users in general?

A substantial part of model design uses instrumentation beyond simple scalar values over time.

> #### Why is this use case valuable to support for your project(s) or organization?

I monitor the dynamics of confusion matrices during training.

> #### Why is it currently difficult to achieve this use case?

The current workaround is to save plots as separate files which is inconvenient in terms of navigation. One cannot compare multiple figures from a run or across runs simultaneously.

### Details

Here is the ui for figures in tensorboard for example:

![Capture d’écran du 2022-07-13 13-35-57](https://user-images.githubusercontent.com/3764009/178724799-122a81b2-f6a9-47e6-9db0-50de7262e963.png)

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
Related to https://github.com/mlflow/mlflow/issues/2696

**Comment by harupy** [comment]
We don't have an immediate plan to support this feature, but we'll keep this issue open to see what the community thinks.

**Comment by kemics** [comment]
@harupy kind reminder about this feature because of `state` label. This would be a very useful feature

**Comment by myaseen-oraclase** [comment]
@nlgranger thanks for opening this issue. I needed something like this in MLFlow for my project and found this issue.

@harupy I'd like to add my support for this feature / enhancement as well. In addition to the confusion matrix example already discussed, I think this feature will also be very useful in a large variety of generative AI tasks where it is invaluable to be able to see the "evolution" of an image etc. over iterations.

**Comment by hyenal** [comment]
I would like to add support for this feature as well for the reason mentioned by @myaseen-oraclase. I work with generative models and this would make evolution tracking easier. This request may be linked to https://github.com/mlflow/mlflow/issues/3843 as well although I feel more strongly about this particular one

**Comment by mcpeixoto** [comment]
I would like to express my support for this feature and strongly believe that it should be implemented in MLflow. TensorBoard already offers this functionality, and it would be highly beneficial for MLflow to incorporate it as well. Tracking the evolution of a model is crucial, particularly for image-based models, and having this capability would greatly enhance the overall usability and effectiveness of MLflow.

**Comment by arvinbalak** [comment]
Yes please! +1 for this issue. Comparisons are not always based on scalar metrics in many cases

**Comment by Antoine101** [comment]
I would like to express my interest for this feature as well. @nlgranger I would like to contribute to it if you're still interested you could look at it together?

**Comment by nlgranger** [comment]
@Antoine101 sorry but I'm no longer using mlflow.

**Comment by simonebonato** [comment]
I would also like to have this feature! I am currently exploring alternative to `WandB`, and I mostly work with object detection and semantic segmentation.

I saw it is possible to upload `plotly` figure or in alternative html files where I can toggle between the input image the GT and the prediction (either masks or masks and boxes), but I would like to see how the predictions evolve over time, hence the need for the `step`. 
This is extremely valuable for me and I can do that in `WandB`


Please keep it up!
